core/templates/pages/admin-page/jobs-tab/admin-jobs-tab.directive.ts

```diff
@@ -41,79 +40,70 @@ angular.module('oppia').directive('adminJobsTab', [
       controller: [function() {
         var ctrl = this;
         ctrl.showJobOutput = function(jobId) {
-          var adminJobOutputUrl = UrlInterpolationService.interpolateUrl(
-            ADMIN_JOB_OUTPUT_URL_TEMPLATE, {
-              jobId: jobId
+          AdminBackendApiService.showJobOutputAsync(jobId)
+            .then(function(response) {
+              ctrl.showingJobOutput = true;
+              ctrl.jobOutput = response || [];
+              ctrl.jobOutput.sort();
+              $timeout(function() {
+                document.querySelector('#job-output').scrollIntoView();
+              });
+            }, function(errorResponse) {
+              ctrl.setStatusMessage(
+                'Server error: ' + errorResponse.error.error);
             });
-          $http.get(adminJobOutputUrl).then(function(response) {
-            ctrl.showingJobOutput = true;
-            ctrl.jobOutput = response.data.output || [];
-            ctrl.jobOutput.sort();
-            $timeout(function() {
-              document.querySelector('#job-output').scrollIntoView();
-            });
-          });
         };
 
         ctrl.startNewJob = function(jobType) {
           ctrl.setStatusMessage('Starting new job...');
 
-          $http.post(ADMIN_HANDLER_URL, {
-            action: 'start_new_job',
-            job_type: jobType
-          }).then(function() {
-            ctrl.setStatusMessage('Job started successfully.');
-            window.location.reload();
-          }, function(errorResponse) {
-            ctrl.setStatusMessage(
-              'Server error: ' + errorResponse.data.error);
-          });
+          AdminBackendApiService.startNewJobAsync(jobType)
+            .then(function() {
+              ctrl.setStatusMessage('Job started successfully.');
+              window.location.reload();
+            }, function(errorResponse) {
+              ctrl.setStatusMessage(
+                'Server error: ' + errorResponse.error.error);
+            });
         };
 
         ctrl.cancelJob = function(jobId, jobType) {
           ctrl.setStatusMessage('Cancelling job...');
 
-          $http.post(ADMIN_HANDLER_URL, {
-            action: 'cancel_job',
-            job_id: jobId,
-            job_type: jobType
-          }).then(function() {
-            ctrl.setStatusMessage('Abort signal sent to job.');
-            window.location.reload();
-          }, function(errorResponse) {
-            ctrl.setStatusMessage(
-              'Server error: ' + errorResponse.data.error);
-          });
+          AdminBackendApiService.cancelJobAsync(jobId, jobType)
+            .then(function() {
+              ctrl.setStatusMessage('Abort signal sent to job.');
+              window.location.reload();
+            }, function(errorResponse) {
+              ctrl.setStatusMessage(
+                'Server error: ' + errorResponse.error.error);
+            });
```

Comment:
Can we change the `cancelJobAsync` such that we just have to do this here:
```
function(errorMessage) {
  ctrl.setStatusMessage( 'Server error: ' + errorMessage);
}
```

Ditto for all other functions (if possible)

Reply:
done